### PR TITLE
set backbone as ResNet

### PR DESCRIPTION
**Pull request by ajy8456**
I select backbone network as ResNet, which is often used for 2D-object detection.

It can be used various version of ResNet, such as 'resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152', 'resnext50_32x4d', 'resnext101_32x8d', 'wide_resnet50_2', 'wide_resnet101_2'.

I referenced the official code of torchvision. :https://github.com/pytorch/vision/blob/master/torchvision/models/detection/backbone_utils.py

### CODE REVIEW

File: backbone.ipynb [hunk 1]
```diff
@@ -0,0 +1,191 @@
+{
```
Comment (yycho0108) [inline backbone.ipynb:1]: <p>Can you add a comment for the logic in line 74-76? Looks like we're string parsing the layer parameter prefixed on the training layers to disable gradient computation. Wasn't super clear on first sight.</p><p> </p><p>Furthermore, I'm not sure if <code>requires_grad_</code> is all we need to do. What about <code>BatchNorm</code> statistics? Will they be frozen as well?</p><p> </p><p>*UPDATE:* ^^ Realized we're explicitly using <code>FrozenBatchNorm2d</code>  for this purpose, NVM.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/yycho0108/ai604-video-object-pose/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (yycho0108) [inline backbone.ipynb:1]: <p>If you'd prefer to do this more rigorously we could try to setup a <code>unittest</code>  or <code>pytest</code>  framework. For now, LGTM.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/yycho0108/ai604-video-object-pose/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (yycho0108) [inline backbone.ipynb:1]: <p>No need for <code>main</code> check  since we're working with a notebook (<code>.ipynb</code>).  This would be more appropriate for a python executable script.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/yycho0108/ai604-video-object-pose/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (yycho0108) [inline backbone.ipynb:1]: <p>Overall the layer name composition and channel dimension parsing in line 81-84 feels super brittle, but I guess they always work for <code>ResNet</code>-based  models?</p><p>We might want to add tests to verify the correctness for this function since we have so many hardcoded assumptions here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/yycho0108/ai604-video-object-pose/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Review by yycho0108 — APPROVED** [review]
Overall LGTM. Thanks for the great work. In general, for core library related capabilities I think we should prefer to develop on `.py` files rather than `.ipynb` since we'd most likely want to import them in the scripts. Thoughts? I think it's out of scope for this PR, but we might want to refactor them in the future.
